### DISCUSSION TITLE
use numactl physical binding to cores

### DISCUSSION THREAD

**Discussion by hajes** [ericaltendorf/plotman#877] [Ideas]
use `numactl --physcpubind=0-3` for CPU cores binding

your script runs 20-40 min longer due long known Linux kernel CPU & IO scheduling

I like your plot manager but plots take longer than running all manually with numactl cores binding/affinity

**Comment by altendky** [comment]
I've got a hacked branch to just barely pull of some NUMA locking with madMAx on my system and I've only noticed a few percent better throughput.  But yeah, maybe something will happen there.  I have a Supermicro X9DR3-LN4F+ with a pair of Intel Xeon E5-2690 v2, separate RAM NUMA locked RAM drives, and a RAID0 of 4x Inland Premium 1TB that are shared and all on a single CPU's node.  That last bit has me "concerned" that my numbers aren't really representative, but I also haven't noticed a significant difference between plots on each node.  I would have expected that if the RAID being on one node was an issue that the node it was on would be faster and the other would be slower.  Also, splitting them up will reduce peak bandwidth.  But I may yet get to doing that.

**Reply by hajes** [reply]
I started on Ubuntu 20.04 LTS and out of box performance was crap. I did lots of research how to force NVMe and CPU to max. performance.

Normal nice/renice, CPU affinity doesn't work well on Ryzen 9 3950X...no idea why. So far, I have found only "numactl" core binding to work best. They suggest for SMP to force processes to each CPU with numactl. Linux kernel scheduling doesn't work well as I have found out in threads as old as 15 years and not much has changed.

Intel may be different story though.

I see my plots 6-7h compared to Plotman 7-8h with 7 plots parallel...it is almost 20% slower performance just by CPU affinity

**Reply by altendky** [reply]
Give the 7x parallel I am guessing you are running the Chia Network chiapos plotter.  Have you tried madMAx?

**Reply by hajes** [reply]
yes I prefer official plotter - I do not want any questionable 3rd party SW ;-) It is like farming on Hpool

**Reply by altendky** [reply]
It's not... there's long been plans to integrate it with the official package.
